packages/plugin/src/parser.ts

```diff
@@ -4,20 +4,27 @@ import { GraphQLError, TypeInfo } from 'graphql';
 import { Linter } from 'eslint';
 import { GraphQLESLintParseResult, ParserOptions } from './types';
 import { extractTokens } from './utils';
-import { getSchema } from './schema';
-import { getSiblingOperations } from './sibling-operations';
-import { loadConfigSync, GraphQLConfig } from 'graphql-config';
+import { getSchema, schemaLoaders } from './schema';
+import { getSiblingOperations, operationsLoaders } from './sibling-operations';
+import { loadConfigSync, GraphQLConfig, GraphQLExtensionDeclaration } from 'graphql-config';
 
 export function parse(code: string, options?: ParserOptions): Linter.ESLintParseResult['ast'] {
   return parseForESLint(code, options).ast;
 }
 
+const addCodeFileLoaderExtension: GraphQLExtensionDeclaration = (api) => {
+  schemaLoaders().forEach(loader => api.loaders.schema.register(loader))
+  operationsLoaders().forEach(loader => api.loaders.documents.register(loader))
+  return { name: 'graphql-eslint-loaders' }
+}
+
 export function parseForESLint(code: string, options?: ParserOptions): GraphQLESLintParseResult {
   const gqlConfig: GraphQLConfig | null = options?.skipGraphQLConfig
     ? null
     : loadConfigSync({
         throwOnEmpty: false,
         throwOnMissing: false,
+        extensions: [addCodeFileLoaderExtension]
       });
```

Comment:
If you are running without graphql-config, we cache those things to avoid performance issues. 
I guess we'll need to do it for graphql-config as well. I created a new issue for it: https://github.com/dotansimha/graphql-eslint/issues/233